[Core] Added the option to caluculate precise diagonal speed of a mouse cursor (#21117)
qmk/qmk_firmware: docs/feature_mouse_keys.md

This is the default mode. You can adjust the cursor and scrolling acceleration using the following settings in your keymap’s `config.h` file:

> |Define                          |Default        |Description                                              |
> |--------------------------------|---------------|---------------------------------------------------------|
> |`MOUSEKEY_DELAY`                |10             |Delay between pressing a movement key and cursor movement|
> |`MOUSEKEY_INTERVAL`             |20             |Time between cursor movements in milliseconds            |
> |`MOUSEKEY_MOVE_DELTA`           |8              |Step size                                                |
> |`MOUSEKEY_MAX_SPEED`            |10             |Maximum cursor speed at which acceleration stops         |
> |`MOUSEKEY_TIME_TO_MAX`          |30             |Time until maximum cursor speed is reached               |
> |`MOUSEKEY_WHEEL_DELAY`          |10             |Delay between pressing a wheel key and wheel movement    |
> |`MOUSEKEY_WHEEL_INTERVAL`       |80             |Time between wheel movements                             |
> |`MOUSEKEY_WHEEL_MAX_SPEED`      |8              |Maximum number of scroll steps per scroll action         |
> |`MOUSEKEY_WHEEL_TIME_TO_MAX`    |40             |Time until maximum scroll speed is reached               |
> |`MOUSEKEY_PRECISE_DIAGONAL_MOVE`|**Not defined**|Precisely caluculated diagonal cursor speed              |
> 

Reviewer: ````suggestion
|Define                          |Default|Description                                              |
|--------------------------------|-------|---------------------------------------------------------|
|`MOUSEKEY_DELAY`                |10     |Delay between pressing a movement key and cursor movement|
|`MOUSEKEY_INTERVAL`             |20     |Time between cursor movements in milliseconds            |
|`MOUSEKEY_MOVE_DELTA`           |8      |Step size                                                |
|`MOUSEKEY_MAX_SPEED`            |10     |Maximum cursor speed at which acceleration stops         |
|`MOUSEKEY_TIME_TO_MAX`          |30     |Time until maximum cursor speed is reached               |
|`MOUSEKEY_WHEEL_DELAY`          |10     |Delay between pressing a wheel key and wheel movement    |
|`MOUSEKEY_WHEEL_INTERVAL`       |80     |Time between wheel movements                             |
|`MOUSEKEY_WHEEL_MAX_SPEED`      |8      |Maximum number of scroll steps per scroll action         |
|`MOUSEKEY_WHEEL_TIME_TO_MAX`    |40     |Time until maximum scroll speed is reached               |

Diagonal movement speed calculation may not be accurate when step size and maximum cursor speed are configured to low values. To improve the accuracy of diagonal movement calculation, add the following in your keymap's `config.h` file:
```c
#define MOUSEKEY_PRECISE_DIAGONAL_MOVE
```

````

Documenting a simple define in its own section will be clearer than adding it to a table of definition with values.

This is the default mode. You can adjust the cursor and scrolling acceleration using the following settings in your keymap’s `config.h` file:

> |Define                          |Default|Description                                              |
> |--------------------------------|-------|---------------------------------------------------------|
> |`MOUSEKEY_DELAY`                |10     |Delay between pressing a movement key and cursor movement|
> |`MOUSEKEY_INTERVAL`             |20     |Time between cursor movements in milliseconds            |
> |`MOUSEKEY_MOVE_DELTA`           |8      |Step size                                                |
> |`MOUSEKEY_MAX_SPEED`            |10     |Maximum cursor speed at which acceleration stops         |
> |`MOUSEKEY_TIME_TO_MAX`          |30     |Time until maximum cursor speed is reached               |
> |`MOUSEKEY_WHEEL_DELAY`          |10     |Delay between pressing a wheel key and wheel movement    |
> |`MOUSEKEY_WHEEL_INTERVAL`       |80     |Time between wheel movements                             |
> |`MOUSEKEY_WHEEL_MAX_SPEED`      |8      |Maximum number of scroll steps per scroll action         |
> |`MOUSEKEY_WHEEL_TIME_TO_MAX`    |40     |Time until maximum scroll speed is reached               |

Reviewer: ```suggestion
|Define                      |Default|Description                                              |
|----------------------------|-------|---------------------------------------------------------|
|`MOUSEKEY_DELAY`            |10     |Delay between pressing a movement key and cursor movement|
|`MOUSEKEY_INTERVAL`         |20     |Time between cursor movements in milliseconds            |
|`MOUSEKEY_MOVE_DELTA`       |8      |Step size                                                |
|`MOUSEKEY_MAX_SPEED`        |10     |Maximum cursor speed at which acceleration stops         |
|`MOUSEKEY_TIME_TO_MAX`      |30     |Time until maximum cursor speed is reached               |
|`MOUSEKEY_WHEEL_DELAY`      |10     |Delay between pressing a wheel key and wheel movement    |
|`MOUSEKEY_WHEEL_INTERVAL`   |80     |Time between wheel movements                             |
|`MOUSEKEY_WHEEL_MAX_SPEED`  |8      |Maximum number of scroll steps per scroll action         |
|`MOUSEKEY_WHEEL_TIME_TO_MAX`|40     |Time until maximum scroll speed is reached               |
```
Restoring original table width.

Diagonal movement speed calculation may not be accurate when step size and maximum cursor speed are configured to low values. To improve the accuracy of diagonal movement calculation, add the following in your keymap's `config.h` file:
```c
#define MOUSEKEY_PRECISE_DIAGONAL_MOVE
```

> 
> Under the method above, the result is always rounded toward zero. So for pursuing most precision, add the following line to `config.h`. This computes the nearest integer value and rounds halfway cases away from zero.
> ```c
> #define MOUSEKEY_PRECISE_DIAGONAL_MOVE_2
> ```

Reviewer: ```suggestion
```
